apps/aecore/src/aec_tx.erl

```diff
@@ -68,20 +72,33 @@ is_coinbase(Signed) ->
 %%% Internal functions
 %%%=============================================================================
 
-apply_on_state_trees(SignedTxs, Trees, Height) ->
-    apply_on_state_trees(SignedTxs, [], Trees, Height).
+apply_signed_common(SignedTxs, Trees0, Height, Strict) ->
+    Trees1 = aec_trees:perform_pre_transformations(Trees0, Height),
```

Comment:
I see - this means that at each and every block the oracles state tree must be pruned.

Reply:
Yes, and the cache we keep make that extremely cheap unless pruning actually has to be performed.

Reply:
There is a cache to make this efficient. Finding the next node that needs to be pruned is O(1).